### PR TITLE
Simplify map provider

### PR DESCRIPTION
**Pull request by marcjansen**
### Description:

This makes it easier for people who use `npm link`, where `instanceof` checks sometimes fail.

Also the code is easier to read. I was unsure about how I handle the tests, but this works locally for me. Looking forward to your review.

### CODE REVIEW

File: src/HigherOrderComponent/MapProvider/MapProvider.spec.jsx [hunk 1]
```diff
@@ -33,16 +33,32 @@ describe('MapProvider', () => {
 
     it('provides a given map to its children (ol.Map)', () => {
       const map = TestUtil.createMap();
+      // We create the promise only to be able to return it below, we do not
+      // actually use it to ibnstantiate MapProvider
```
Comment (KaiVolland) [inline src/HigherOrderComponent/MapProvider/MapProvider.spec.jsx]: typo

File: src/HigherOrderComponent/MapProvider/MapProvider.jsx [hunk 2]
```diff
@@ -54,19 +54,13 @@ class MapProvider extends React.Component {
       ready: false
     };
 
-    if (props.map instanceof OlMap) {
-      this.state = {
-        map: props.map,
+    Promise.resolve(props.map).then((map) => {
```
Comment (dnlkoch) [inline src/HigherOrderComponent/MapProvider/MapProvider.jsx]: What happens if `props.map` is being rejected? Could you provide a test?

Comment (marcjansen) [reply src/HigherOrderComponent/MapProvider/MapProvider.jsx]: Good question. On it right now.

**Comment by marcjansen** [comment]
@dnlkoch I have added 630457836de977448dc97a3a44fb5129e78a37a5; I hope this *resolves* (pun very much intended) this discussion.

File: src/HigherOrderComponent/MapProvider/MapProvider.jsx (hunk 2 above)
Comment (marcjansen) [reply src/HigherOrderComponent/MapProvider/MapProvider.jsx]: See https://github.com/terrestris/react-geo/pull/373/commits/8efe62e6ea201d485bf45eb866656a3a196355d3

Comment (dnlkoch) [reply src/HigherOrderComponent/MapProvider/MapProvider.jsx]: Awesome! :+1:

**Comment by dnlkoch** [comment]
@marcjansen Thanks again for improvement, please merge.

**Comment by marcjansen** [comment]
Thanks for the comments, guys.

**Pull request merged**